nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -138,6 +220,51 @@ def __call__(self, imgs):
         return signals, None
 
 
+class _InversionFunctor(object):
+
+    func_name = 'nifti_spheres_masker_inversion'
+
+    def __init__(self, seeds_, radius, mask_img, allow_overlap, dtype):
+        self.seeds_ = seeds_
+        self.radius = radius
+        self.mask_img = mask_img
+        self.allow_overlap = allow_overlap
+        self.dtype = dtype
+        # Mask is only used to get affine and shape
+        if self.mask_img is not None:
+            mask = check_niimg_3d(self.mask_img)
+            self.target_affine = mask.affine
+            self.target_shape = mask.shape
+        else:
+            # If no mask is provided use the MNI affine and shape
+            self.target_affine = np.array([[  3.,   -0.,   -0.,   -90.],
+                                           [  -0.,    3.,   -0., -126.],
+                                           [   0.,    0.,    3.,  -72.],
+                                           [   0.,    0.,    0.,    1.]])
+            self.target_shape = [61, 73, 61]
+
+    def __call__(self, signals):
+        n_seeds = len(self.seeds_)
+        # np.empty creates unexpected behavior!
+        spheres = np.zeros((np.prod(self.target_shape), n_seeds), dtype=self.dtype)
+
+        for i, sphere in enumerate(_iter_regions_from_spheres(
+                self.seeds_, self.target_affine, self.target_shape, self.radius,
+                self.allow_overlap, mask_img=self.mask_img)):
+            spheres[sphere, i] = 1
+
+        # Compute overlap scaling for mean signal:
+        if self.allow_overlap:
+            spheres_sum = spheres.sum(1)
+            # signals_to_img_maps uses dot product to create mask over regions,
+            # inverse scaling to create average not sum
+            spheres_scale = 1 / spheres_sum[spheres_sum > 1, np.newaxis]
```

Comment:
aren't we going to have divisions by 0 here ?

Reply:
I don't think so. `spheres_sum` is only selection overlapping areas. I.e. areas which sums are > 1.  Only for these points the division is done, definitely open to suggestions, but that seemed the easiest way for me to create the average signal of overlapping voxels.